plugins/woocommerce/src/Blocks/BlockTypes/AddToCartWithOptionsVariationSelectorItemTemplate.php

```diff
@@ -23,4 +25,133 @@ class AddToCartWithOptionsVariationSelectorItemTemplate extends AbstractBlock {
 	protected function get_block_type_style() {
 		return null;
 	}
+
+	/**
+	 * Render the block.
+	 *
+	 * @param array    $attributes Block attributes.
+	 * @param string   $content Block content.
+	 * @param WP_Block $block Block instance.
+	 * @return string Rendered block output.
+	 */
+	protected function render( $attributes, $content, $block ): string {
+		global $product;
+
+		$content = '';
+
+		$product_attributes = $product->get_variation_attributes();
+
+		foreach ( $product_attributes as $product_attribute_name => $product_attribute_terms ) {
+			$content .= $this->get_product_row( $product_attribute_name, $product_attribute_terms, $attributes, $block );
+		}
+
+		return $content;
+	}
+
+	/**
+	 * Get product row HTML.
+	 *
+	 * @param string   $product_attribute_name Product Attribute Name.
+	 * @param array    $product_attribute_terms Product Attribute Terms.
+	 * @param array    $attributes Block attributes.
+	 * @param WP_Block $block The Block.
+	 * @return string Row HTML
+	 */
+	private function get_product_row( $product_attribute_name, $product_attribute_terms, $attributes, $block ): string {
+		$attribute_name  = $product_attribute_name;
+		$attribute_terms = $this->get_terms( $product_attribute_name, $product_attribute_terms );
+
+		if ( empty( $attribute_terms ) ) {
+			return '';
+		}
+
+		// Get an instance of the current Post Template block.
+		$block_instance = $block->parsed_block;
+
+		$new_block = new WP_Block(
+			$block_instance,
+			array(
+				'attributeId'    => uniqid(),
```

Comment:
I think HTML element ids shouldn't start with a number, and `uniqid()` can return values like `67d1b217a8603`. So we might want to prefix a string before the id.

Reply:
done